File: camerax_android/android/src/main/kotlin/dev/zeekr/camerax_android/video/QualityImpl.kt
package dev.zeekr.camerax_android.video

import androidx.camera.video.Quality
import dev.zeekr.camerax_android.CameraXApiPigeonProxyApiRegistrar
import dev.zeekr.camerax_android.PigeonApiQualityProxyApi

class QualityImpl(registrar: CameraXApiPigeonProxyApiRegistrar) : PigeonApiQualityProxyApi(registrar) {
    override fun fhd(): Quality {
        return Quality.FHD
    }

    override fun hd(): Quality {
        return Quality.HD
    }

    override fun highest(): Quality {
        return Quality.HIGHEST
    }

    override fun lowest(): Quality {
        return Quality.LOWEST
    }

    override fun sd(): Quality {
        return Quality.SD
    }

    override fun uhd(): Quality {
        return Quality.UHD
    }
}

File: camerax_ios/lib/src/video/quality_impl.dart
// import 'package:camerax_ios/src/api.dart';
// import 'package:camerax_platform_interface/camerax_platform_interface.dart';

// final class QualityImpl implements Quality {
//   final QualityProxyApi api;

//   QualityImpl.internal(this.api);
// }

// final class QualityChannelImpl extends QualityChannel {
//   @override
//   Quality get sd => QualityImpl.internal(QualityProxyApi.sd);
//   @override
//   Quality get hd => QualityImpl.internal(QualityProxyApi.hd);
//   @override
//   Quality get fhd => QualityImpl.internal(QualityProxyApi.fhd);
//   @override
//   Quality get uhd => QualityImpl.internal(QualityProxyApi.uhd);
//   @override
//   Quality get lowest => QualityImpl.internal(QualityProxyApi.lowest);
//   @override
//   Quality get highest => QualityImpl.internal(QualityProxyApi.highest);
// }

// extension QualityX on Quality {
//   QualityProxyApi get api {
//     final impl = this;
//     if (impl is! QualityImpl) throw TypeError();
//     return impl.api;
//   }
// }

// extension QualityProxyApiX on QualityProxyApi {
//   Quality get impl => QualityImpl.internal(this);
// }
